### PR TITLE
Added development section to the readme

### PR DESCRIPTION
**Pull request by MohamedNasser8**
In this PR I added the following:

- Development Section: A new section was added to the README file to guide developers.
- Environment Options: Provided an option for users to choose between Poetry or venv for managing dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,80 @@ We welcome contributions to OSIPI! To contribute, follow these steps:
...
+2. You can run all pre-commit hooks on all files with:
+
+    ```bash
+    pre-commit run --all-files
+    ```
+
+After this, `pre-commit` will automatically run the hooks each time you commit changes to the project. If any of the hooks fail,
+you'll need to fix the issues and then try committing your changes again.
+
+Also some of hooks are auto fixed if you see file was modified after running `pre-commit run --all-files` then you can add the changes and commit again.
```
Comment (ltorres6) [inline README.md:107]:
```suggestion
NOTE: Some of hooks automatically fix/format your files to specific standards. If you see file was modified after running `pre-commit run --all-files` then you'll need to add the changes (git add) and commit again.
```

File: README.md [hunk 2]
```diff
@@ -33,7 +33,80 @@ We welcome contributions to OSIPI! To contribute, follow these steps:
...
+        source venv/bin/activate
+        ```
+
+5. Install the required packages from `requirements.txt`:
+
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+### Option 2: Using Poetry
```
Comment (ltorres6) [inline README.md:70]: We want to make poetry a requirement. It's small overhead and it ensures packages won't be missing/incompatible. Remove this and the associated requirements.txt please.

**Comment by MohamedNasser8** [comment]
I think it's ready to be merged! @ltorres6

File: README.md [hunk 3]
```diff
@@ -100,11 +82,10 @@ There are two ways to set up your development environment for this project. You
     ```bash
     pre-commit run --all-files
     ```
-
-After this, `pre-commit` will automatically run the hooks each time you commit changes to the project. If any of the hooks fail,
-you'll need to fix the issues and then try committing your changes again.
-
-Also some of hooks are auto fixed if you see file was modified after running `pre-commit run --all-files` then you can add the changes and commit again.
+**NOTE:**
+- Next times you commit changes, `pre-commit` will run the hooks automatically.
```
Comment (ltorres6) [inline README.md:86]:
```suggestion
- Next time you commit changes, `pre-commit` will run the hooks automatically.
```

File: README.md [hunk 4]
```diff
@@ -35,41 +35,23 @@ We welcome contributions to OSIPI! To contribute, follow these steps:
 
 ## Development Setup
 
-There are two ways to set up your development environment for this project. You can choose the one that suits you best.
+We used poetry to manage the dependencies for this project.
```
Comment (ltorres6) [inline README.md:38]:
```suggestion
We use poetry to manage the dependencies for this project.
```

File: README.md [hunk 5]
```diff
@@ -100,11 +82,10 @@ There are two ways to set up your development environment for this project. You
...
     pre-commit run --all-files
     ```
-
-After this, `pre-commit` will automatically run the hooks each time you commit changes to the project. If any of the hooks fail,
-you'll need to fix the issues and then try committing your changes again.
-
-Also some of hooks are auto fixed if you see file was modified after running `pre-commit run --all-files` then you can add the changes and commit again.
+**NOTE:**
+- Next times you commit changes, `pre-commit` will run the hooks automatically.
+- Some of hooks automatically fix/format your files to specific standards. If you see that some of your files have been changed after a commit, don't worry! It's just `pre-commit` doing its job.
```
Comment (ltorres6) [inline README.md:87]:
```suggestion
- Some hooks automatically fix/format your files to specific standards. If you see that some of your files have been changed after a commit, don't worry! It's just `pre-commit` doing its job.
```

Comment (ltorres6) [reply README.md:87]: Nice.

**Comment by ltorres6** [comment]
@MohamedNasser8 Thanks Mohamed, one more iteration for the typos please. I am going to add Lucy and Sirisha to review as well with fresh eyes.

File: README.md
Comment (stadmill) [inline README.md:1]: I can see you've added instructions on setting up the development environment. Perhaps there is good reason for this, but shouldn't these instructions be merged into the developer guide?

Comment (MohamedNasser8) [reply README.md:1]: I think keeping the development environment set in the README for now is important, as it informs anyone coming to the repo on how to start contributing. Once we have a published version, we can update the README accordingly and remove the development setup part. What do you think, @ltorres6 ?

**Comment by MohamedNasser8** [comment]
I think keeping the development environment set in the README for now is important, as it informs anyone coming to the repo on how to start contributing. Once we have a published version, we can update the README accordingly and remove the development setup part. What do you think, @ltorres6 ?

**Comment by ltorres6** [comment]
I think a good compromise is that it should be in both. We should give people a simple overview of how to start developing in the readme for quick start users, and then point to a more detailed version in the docs for users who want/need to know more. This would maintain visibility/exposure of our guidelines from the repo, and allow us to expand in as much detail as we want/need within the docs.

**Comment by ltorres6** [comment]
@MohamedNasser8 Can you add some details on the pre-commit hooks?

**Comment by MohamedNasser8** [comment]
Okay, will do it

**Review by ltorres6 — APPROVED** [review]
LGTM

**Pull request merged**